src/streamlink/plugins/live_russia_tv.py

```diff
@@ -45,37 +46,44 @@ def _get_stream_info_url(self, url):
     def _get_streams(self):
         self.session.http.headers.update({"User-Agent": useragents.FIREFOX})
         iframe_url = self._get_iframe_url(self.url)
+        live = self._live_re.match(self.url)
 
         if iframe_url:
             log.debug("Found iframe URL={0}".format(iframe_url))
             info_url = self._get_stream_info_url(iframe_url)
-
-            if info_url:
-                log.debug("Getting info from URL: {0}".format(info_url))
-                res = self.session.http.get(info_url, headers={"Referer": iframe_url})
-                data = self.session.http.json(res)
-
-                if data['status'] == 200:
-                    for media in data['data']['playlist']['medialist']:
-                        if media['errors']:
-                            log.error(media['errors'].replace('\n', '').replace('\r', ''))
-
-                        for media_type in media.get('sources', []):
-
-                            if media_type == "m3u8":
-                                hls_url = media['sources'][media_type]['auto']
-                                for s in HLSStream.parse_variant_playlist(self.session, hls_url).items():
-                                    yield s
-
-                            if media_type == "http":
-                                for pix, url in media['sources'][media_type].items():
-                                    yield "{0}p".format(pix), HTTPStream(self.session, url)
-                else:
-                    log.error("An error occurred: {0}".format(data['errors'].replace('\n', '').replace('\r', '')))
+        elif live:
```

Comment:
some dupplicate code of `_get_stream_info_url`
fix it there instead of creating it twice